StortSpelProjekt/Game/src/Gamefiles/Shop.h

```diff
@@ -54,8 +56,9 @@ class Shop
 	void OnShopGUIStateChange(shopGUIStateChange* collisionEvent);
 
 private:
-	void upgradePressed(ButtonPressed* evnt);
+	void shopButtonPressed(ButtonPressed* evnt);
 	void sceneChange(SceneChange* evnt);
+	void updateShop();
```

Comment:
Other name than updateShop(), sounds like it should refresh the ui/logic of the shop but it also randomizes the inventory of the shop.